cockatrice/src/window_main.cpp

```diff
@@ -261,7 +261,7 @@ void MainWindow::loginError(Response::ResponseCode r, QString reasonStr, quint32
             break;
         }
         case Response::RespUsernameInvalid:
-            QMessageBox::critical(this, tr("Error"), tr("Invalid username."));
+            QMessageBox::critical(this, tr("Error"), tr("Invalid username.\nAvailable chars: [a-z][A-Z][0-9][-_]"));
```

Comment:
Overlooked this one in your second commit? ;)

btw:
Isn't the "," after "_" in **[...] 0-9, _, and - [...]** wrong? 
"Invalid username.\nYou may only use A-Z, a-z, 0-9, _ and - in your username." (without the "," after "_")